src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/JavaCallableWrapperGenerator.cs

```diff
@@ -855,16 +855,21 @@ void WriteInstrumentationOnCreate (TextWriter sw, Action<TextWriter> extra)
 		}
 
 		StreamWriter OpenStream (string outputPath)
+		{
+			string destination = GetDestinationPath (outputPath);
+			Directory.CreateDirectory (Path.GetDirectoryName (destination));
+			return new StreamWriter (new FileStream (destination, FileMode.Create, FileAccess.Write));
+		}
+
+		/// <summary>
+		/// Returns a destination file path based on the package name of this Java type
+		/// </summary>
+		public string GetDestinationPath (string outputPath)
 		{
 			string path = outputPath;
 			foreach (string dir in package.Split ('.'))
```

Comment:
Given the desire to go "in memory" to reduce disk access, should we also consider trying to reduce *memory* usage?

Instead of `package.Split('.')`, we could use `StringBuilder`:

```csharp
var path = new StringBuilder (outputPath)
    .Append (outputPath);
buf.Replace ('.', Path.DirectorySeparatorChar);
buf.Append (name);
buf.Append (".java");
return but.ToString ();
```

Reply:
I actually found I didn't want to use `StringBuilder` here, I like how `Path.Combine` does the work of handling trailing/ending separators automatically.

But using `string.Replace` and a single `Path.Combine`, it does seem like we'll get better perf than what was here originally.

`Path.Combine` looks like it uses `StringBuilder` internally: https://referencesource.microsoft.com/#mscorlib/system/io/path.cs,1265